### PR TITLE
Model Changes for UUID instead of Object holding. 

### PR DESCRIPTION
**Pull request by RyanKusters**
I have not fixed all the breaks yet, I am looking for feedback about direction. will continue working.

### CODE REVIEW

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/model/Answer.java [hunk 1]
```diff
@@ -71,15 +71,14 @@ public void setAuthor(String author) {
...
+	public boolean hasComment(UUID comment) {
 		return commentList.contains(comment);
 	}
 
 	//add comment if not already added
-	public void addComment(Comment<Answer> comment) {
+	public void addComment(UUID comment) {
 		if (!hasComment(comment)) {
 			commentList.add(comment);
-			comment.setParent(this);
```
Comment (stephenjust) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/model/Answer.java:82]: Why did this disappear?

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/model/Question.java [hunk 2]
```diff
@@ -41,21 +41,20 @@ public Question(String title, String body, String author, Image image) {
...
+	public void addAnswer(UUID a) {
 		if (!answerList.contains(a)) {
 			answerList.add(a);
-			a.setParent(this);
 		}
 	}
 	
-	public boolean hasAnswer(Answer a) {
+	public boolean hasAnswer(UUID a) {
 		return answerList.contains(a);
```
Comment (stephenjust) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/model/Question.java:58]: No reason to change this signature. Just return `answerList.contains(a.getId())`.
Or overload so both work. No use creating extra work by making you update every reference to this if you can make an easy change.

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/model/Question.java [hunk 3]
```diff
@@ -97,15 +96,14 @@ public void setId(UUID id) {
 		this.id = id;
 	}
 	
-	public boolean hasComment(Comment<Question> comment) {
+	public boolean hasComment(UUID comment) {
```
Comment (stephenjust) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/model/Question.java:99]: Same as above.

**Comment by stephenjust** [comment]
Ignore some of my comments. I forgot that you can't turn a UUID into the object it refers to  without calling DataManager, which means you need Context too. Stupid Android. Models should not need to know context.

It's OK for "set methods" to accept objects and extract UUIDs for the data structure (or accept both UUID and object for minimal impact on other code by using overloading) but return UUIDs in "get methods". Models should probably not be calling DataManager.

@Kopo101, what are your thoughts? @RyanKusters, what sounds easier?
I did not mean for this to be a wide reaching design change - as little external impact as possible.

That's a long comment. Contact me if my 11pm disjoint thoughts make no sense.

**Comment by Cotto4** [comment]
Dammit, I was typing out a long comment and accidently hit back. Lost it all. 

Basically what I'm thinking is that Ryan has a solid idea right now. Models should never call DM, in my opinion anyway. Unless I'm misreading this, a lot of the work that has been done has minimized the amount of UUIDs required for a GET call, which should make ES pretty happy. Plus, since we can't call DM without a context, this would save us from passing around the context everywhere the model travels.

Let me know if I'm out to lunch, but I like how Ryan is approaching it so far.

**Comment by stephenjust** [comment]
I deleted the contradictory comments from 10pm Stephen. There are things I like, and places where I think you're making the change harder than it needs to be. 11pm Stephen still wasn't much more sensible, so 12:30am Stephen is hopefully an improvement. It's like the Ballmer peak, but with levels of exhaustion.

In general I like the change set, but there are a few places where you can avoid breaking existing code using overloading. So, defining both `hasComment(UUID commentId)` and `hasComment(Comment<T> comment)` for example. Both are one-liners. If it's easy to update all references, then don't worry about this.

I think the most work fixing other stuff comes from the DataManager interface change, and in that case, you can use Eclipse tools to do all of the annoying refactoring for you because you're just removing parameters, not adding.

**Comment by stephenjust** [comment]
Superseded by #127

**Pull request closed without merging**